FIX: compatibility with Python v2.x

nnako opened the pull request:
modified some bits in order to make the code run on Python v2.x (checked with v2.7). Imported the standard module "io" which provides unicode character handling for Python v2 and v3 in the same way. Additionally, the call of the function "super()" had to be modified in order to fit both Python versions. Some cosmetics regarding the function comments were made to facilitate better standard folding within editors.

nnako committed:
FIX: compatibility with Python v2.x
MOD: some cosmetics regarding folding of function comments

EpocDotFr commented on todotxtio.py:
```diff
@@ -1,5 +1,9 @@
+#!/usr/bin/python
```
This file isn't meant to be executable. What is the purpose of this line?

nnako replied:
hmmm... I always create my librarys so that in the end, they can be used as libraries AND can be executed as standalone tools (where this makes sense). So, you are right, here the line makes no sense, yet.

EpocDotFr commented on todotxtio.py:
```diff
@@ -56,26 +68,43 @@ def from_file(file_path, encoding='utf-8'):
     :rtype: list
     """
+
+    # init
```
I'm sorry but the comments you added in this function doesn't add any useful information, given the statements are simple enough to understand. Please remove them.

nnako replied:
you are right. the comments don't add any functionality. It is just a question of style and readability in general (which is in the end a question of personal taste). I can remove it.

EpocDotFr commented:
Hi, thank you for your interest in Todo.txt I/O!

I added some comments in your PR. Once they'll be addressed, this PR will be ready to be merged and a new version of Todo.txt I/O will be released. Also please create a contributors list in the readme (like [here](https://github.com/EpocDotFr/webtodotxt#contributors)) then add yourself as well as what you done in a few words.

EpocDotFr commented:
Also please update the docs (saying it's also compatible with Python 2) as well as the setup.py.

nnako committed:
FIX: the CREATION DATE will always appear in the 4th group of the defined regex

EpocDotFr commented on todotxtio.py:
```diff
@@ -85,43 +114,57 @@ def from_string(string):
                     todo.completion_date = todo_pre_data.group(2)
             else:
-                todo.creation_date = todo_pre_data.group(2)
+                todo.creation_date = todo_pre_data.group(4)
```
Well that's working ATM, maybe the regex is wrong. Better check the regex first. Anyway I never had issues with creation dates.

nnako replied:
please check with the following code (just open the Python interpreter and paste the code in. I did it for both, Python v2.7 and Python v3.6):

```python
import re
todo_data_regex = re.compile( \
    '^(?:(x) )?' + \
    '(?:(\d{4}-\d{2}-\d{2}) )?' + \
    '(?:\(([A-Z])\) )?' + \
    '(?:(\d{4}-\d{2}-\d{2}) )?' \
    )
match = todo_data_regex.match( \
    '(A) 2018-03-28 an unfinished task created at 2018-03-28 +TestProject @Home due:2018-03-29')
print( \
    'the complete prefix: ' + str(match.group(0)) + '\n' + \
    'the COMPLETED FLAG:  ' + str(match.group(1)) + '\n' + \
    'the COMPLETED DATE:  ' + str(match.group(2)) + '\n' + \
    'the PRIORITY:        ' + str(match.group(3)) + '\n' + \
    'the CREATION DATE:   ' + str(match.group(4)) \
    )
```

in both versions, the creation date should appear in ```match.group(4)``` . When inserting a COMPLETED TASK in the code above, the creation date appears in ```match.group(4)``` as well. As I understand RegEx there, the algorithm should not differ with a changed input string.

nnako committed:
removed some comments

nnako committed:
update of documentation and Python version info

nnako committed:
NEW: PERSON specifications by using different SQUARE BRACKET formats
NEW: REMARKS specification by using CURLY BRACKETS

Commit:
some minimal comment changes

Commit:
FIX: usage of serialize() instead of __str__() so in Python v2 unicode can be handled
FIX: __setattr__() function now handles persons and remarks as well
MOD: rename "author" -> "authors" and "responsible" -> "responsibles" as they are lists

Commit:
FIX: facilitate normal time string "12:34" within text by forcing starting alpha char for attributes

Commit:
FIX: occasional creation date mismatch corrected on import of TODOTXT file

Commit:
FIX: return REMARKS not as list but as string containing '\n'

Commit:
FIX: remove warning message for REMARK to be no string

Commit:
FIX: evaluate REMARKS before PERSONS to facilitate square brackets within remarks

Commit:
NEW: prioritized evaluation of specific link attributes (http, https,
link) into list of list

Commit:
FIX: for detection of tags, take tag name until FIRST colon char

Commit:
added better comments

Commit:
FIX: ignore single PLUS symbol for project specification (while reading from file)

Commit:
FIX: change order of member detection. now, e.g. projects are not read from descriptions.

nnako committed:
todotxt-v1.1.2

nnako committed:
FIX: stabilize project / subject identification against special characters instilled e.g. by GitLab
     special characters following the plus sign will skip the element

nnako committed:
FIX: use recursive regular expressions for identifying remarks correctly

nnako committed:
FIX: stabile detection of tagged values in todotxt texts. omit empty values, backslash at start, ...

nnako committed:
FIX: syntax warning announced by Python 3.12 in strings
FIX: some minor specialities concerning reasonable values

nnako committed:
some better comments and new version

nnako committed:
FIX: distinguish clearly between responsible, author and tracker

nnako committed:
MOD: shift source into approptiate folder

nnako committed:
MOD: rename main source file

nnako committed:
FIX: re-work regular expressions to better cover real usecases

nnako committed:
FIX: rename todotxtio into todotxt as this better resembles general naming conventions

nnako committed:
FIX: ensure no newlines within a todotxt task element. even when the user uses these.

in this todotxt module, a BACKSLASH character will be the textual representation of a newline character. thus, the concept "one text line per todotxt item" stays respected but the user has the ability to (at least in a very basic way) structurize the text.

nnako committed:
FIX: better build behaviour when using "editable"
